### ISSUE TITLE
What are loss results to be expected and at what step, and how is or can validation/testing be done?

### ISSUE THREAD

**Issue by brb1901** [nilboy/tensorflow-yolo#9]
This looks like nice work! But I'm having some trouble interpreting the results and seeing how you do (and I could do) validation/testing (on validation/test set).
When I follow the instructions for training on the Pascal VOC dataset, exactly as you describe it, the system seems to start training fine, and progress can for example be monitored using TensorBoard (or just watching the command line output). 
After about 14000 steps, I get:
```
2017-03-08 13:54:49.951253: step 13960, loss = 4.10 (49.3 examples/sec; 0.325 sec/batch)
2017-03-08 13:54:53.520895: step 13970, loss = 5.73 (45.7 examples/sec; 0.350 sec/batch)
2017-03-08 13:54:57.180532: step 13980, loss = 6.08 (37.5 examples/sec; 0.427 sec/batch)
2017-03-08 13:55:00.669307: step 13990, loss = 4.97 (50.1 examples/sec; 0.319 sec/batch)
2017-03-08 13:55:04.188500: step 14000, loss = 4.71 (50.0 examples/sec; 0.320 sec/batch)
2017-03-08 13:55:07.815220: step 14010, loss = 5.66 (44.7 examples/sec; 0.358 sec/batch)
```
But this is similar to the loss performance observed after only around 1300 steps or so.
What is the loss performance I should expect, and at how many steps (more or less)? 
Is the conf/train.cfg you provide the exact same config file you used to obtain good performance?

And how do you do validation/testing on a separate dataset? Is there a separate function to call or should train.py be used with different parameters or something?

**Comment by nilboy** [comment]
I encounter the same problem when training  from scratch. The training code maybe have some bugs. But,  I can't fix it currently.

**Comment by mldm4** [comment]
@brb1901 Did you find yourself an answer?

**Comment by brb1901** [comment]
@mldm4 No, I gave up on this repository a while ago. You may want to check out Darkflow; or Tensorflow's Object Detection API, which includes reasonable alternatives like SSD.
